### PR TITLE
Validate language & preview picture fields

### PR DESCRIPTION
**Pull request by j0k3r**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | no
| Fixed tickets | https://github.com/wallabag/wallabag/issues/3186
| License       | MIT

_This is a proposal.
Any suggestions on which fields might be a good to validate is welcome._

Instead of saving the value of each field right into the content without any validation, it seems better to validate them **before**.
This might sounds obvious now we say that but we didn't do it before that PR.

### CODE REVIEW

**Comment by nicosomb** [comment]
Nice 👍

**Comment by nicosomb** [comment]
> Any suggestions on which fields might be a good to validate is welcome.

For me, it looks OK.

File: src/Wallabag/CoreBundle/Helper/ContentProxy.php [hunk 1]
```diff
@@ -7,6 +7,9 @@
 use Wallabag\CoreBundle\Entity\Entry;
 use Wallabag\CoreBundle\Tools\Utils;
 use Symfony\Component\HttpFoundation\File\MimeType\MimeTypeExtensionGuesser;
+use Symfony\Component\Validator\Constraints\Language as LanguageConstraint;
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Helper/ContentProxy.php]: Language is `en` and not `en_GB` for instance. However, the OpenGraph `og:locale` format is `language_TERRITORY`, so it should be `Symfony\Component\Validator\Constraints\Locale` instead.

Handling merging `en_GB` and `en_US` into `en` should be done somewhere else.

Comment (nicosomb) [inline src/Wallabag/CoreBundle/Helper/ContentProxy.php]: You can also do this for `fr_FR`, `fr_*`, etc.

Comment (j0k3r) [inline src/Wallabag/CoreBundle/Helper/ContentProxy.php]: Hum, I'll check that.
We don't use the `og:locale` as a fallback for the language field return by graby.

**Comment by tcitworld** [comment]
This has a breaking change : if language or preview_picture are not found, it returns `''` instead of `null`. Therefore it saves an empty string in the database for nothing, templates show empty pictures (since we check for `null` and not `empty`) and filters don't work anymore.

**Pull request merged**